script.js

```diff
@@ -1,28 +1,98 @@
-//You can edit ALL of the code here
+
+let url = `https://api.tvmaze.com/shows/1632/episodes`;
+let selectedShow = 0;
 let episodeLength;
-function setup() {
-  createSelectEpisode();
-  createSearchSection();
-  const allEpisodes = getAllEpisodes();
-  episodeLength = allEpisodes.length;
-  makePageForEpisodes(allEpisodes);
-  searchElement = document.getElementById("searchKey");
-  searchElement.addEventListener("input", search);
-  let SelectElement = document.getElementById("selectEpisode");
-}
-function createEpisodeCode(number, season) {
-  number = number < 10 ? `0${number}` : number;
-  season = season < 10 ? `0${season}` : season;
-  return `S${number}E${season}`;
+let allEpisodes = [];
+
+function setup(url, selectedShow) {
+  makeSectionForSelects();
+  fillSelectByShows(selectedShow);
+  fetch(url)
+    .then((Response) => Response.json())
+    .then(function (data) {
+      allEpisodes = data;
+      episodeLength = allEpisodes.length;
+      makeSearchSection();
+      fillSelectByEpisodes(allEpisodes);
+      makeSectionForEpisodes(allEpisodes);
+    })
+    .catch((error) => console.error(error));
 }
-function createSelectEpisode() {
+
+function makeSectionForSelects() {
+  const rootElem = document.getElementById("root");
   let searchSection = document.createElement("section");
-  searchSection.id = "searchBox";
-  searchSection.className = "row p-4 ";
-  let SelectElement = document.createElement("select");
-  SelectElement.className = "col-4 m-3 p-2 form-control";
-  SelectElement.id = "selectEpisode";
-  let allEpisodes = getAllEpisodes();
+  searchSection.id = "searchSection";
+  searchSection.className = "row p-4 mx-auto";
+  rootElem.append(searchSection);
+  let showSelectElement = document.createElement("select");
+  showSelectElement.id = "showSelect";
+  showSelectElement.className = "col-10 mx-auto p-2";
+  let episodeSelectElement = document.createElement("select");
+  episodeSelectElement.className = "col-3 m-3 p-2 form-control";
+  episodeSelectElement.id = "selectEpisode";
+  searchSection.append(
+    showSelectElement,
+    episodeSelectElement
+  );
+}
+
+function fillSelectByShows(selectedShow) {
+  let searchSection = document.getElementById("searchSection");
+  let SelectElement = document.getElementById("showSelect");
+  let allShows = getAllShows();
+  allShows.map((show) => {
```

Comment:
@DibaKarimi  instead of using the id of each episode for the value of the options you can use a map with it's index ,
you can write down map((show,index) => .....) ,
check this : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map
what it does , if you inspect your code you will see in different options you have weird values ( because they are the ids of the episodes) , in this way you will have values from 1 and they are in order ( 1, 2, 3, ....)